vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/devtools/SeleniumMobileEmulationSteps.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2019-2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.steps.ui.web.devtools;
+
+import com.google.common.eventbus.EventBus;
+
+import org.vividus.steps.ui.web.MobileEmulationSteps;
+import org.vividus.ui.web.cdp.CdpClient;
+import org.vividus.ui.web.event.DeviceMetricsOverrideEvent;
+
+public class SeleniumMobileEmulationSteps extends MobileEmulationSteps
+{
+    private final EventBus eventBus;
+
+    public SeleniumMobileEmulationSteps(CdpClient cdpClient, EventBus eventBus)
+    {
+        super(cdpClient);
+        this.eventBus = eventBus;
+    }
+
+    @Override
+    protected void afterMetricsOverride()
+    {
+        eventBus.post(new DeviceMetricsOverrideEvent());
```

Comment:
move this logic to selenium CDP client: if event name is XYZ, then post `DeviceMetricsOverrideEvent`